pyiceberg/io/fsspec.py

```diff
@@ -124,6 +128,22 @@ def _file(_: Properties) -> LocalFileSystem:
     return LocalFileSystem(auto_mkdir=True)
 
 
+def _oss(properties: Properties) -> AbstractFileSystem:
+    from s3fs import S3FileSystem
+
+    client_kwargs = {
+        "endpoint_url": properties.get(OSS_ENDPOINT),
+        "aws_access_key_id": properties.get(OSS_ACCESS_KEY_ID),
+        "aws_secret_access_key": properties.get(OSS_ACCESS_KEY_SECRET),
+        "aws_session_token": properties.get(OSS_SESSION_TOKEN),
+    }
+    config_kwargs = {"s3": {"addressing_style": "virtual"}, "signature_version": "v4"}
```

Comment:
Should wel also make virtual addressing configurable? Similar to `S3_FORCE_VIRTUAL_ADDRESSING`?
```suggestion
    config_kwargs = {"s3": {"addressing_style": "virtual"}, "signature_version": "v4"}
```

Reply:
Since the init function for OSS is already separated from S3, and users aren't able to interact with OSS without virtual addressing set to true anyway, I think it makes more sense to just set it to true by default to make it simpler.

Reply:
But we've already crossed that bridge with `PyArrowFileIO`: https://github.com/apache/iceberg-python/pull/1788/files#diff-8d5e63f2a87ead8cebe2fd8ac5dcf2198d229f01e16bb9e06e21f7277c328abdL423-L424

I think it would also be good to add a configuration option for it. we don't know if people are using it, and just removing it feels wrong.

Reply:
To be honest, I'm not entirely sure which `config_kwargs` keys supported by s3fs that can work with OSS connection. The documentation for `boto3` compatibility only mentioned setting several keys: https://www.alibabacloud.com/help/en/oss/developer-reference/use-amazon-s3-sdks-to-access-oss#section-jmf-a67-hat

Do you think it'll make sense to copy the `config_kwargs` setting for S3 but with `oss.config-example` parameter?